### PR TITLE
[Java] Serialize OffsetDateTime in UTC timezone

### PR DESCRIPTION
**Pull request by cbornet**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run`./bin/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

See #3439

### CODE REVIEW

**Comment by cbornet** [comment]
I'm first doing it for feign to get your comments. Then I'll do it for the other frameworks.

File: modules/swagger-codegen/src/main/resources/Java/libraries/feign/ApiClient.mustache [hunk 1]
```diff
@@ -28,6 +34,10 @@ import {{invokerPackage}}.auth.OAuth.AccessTokenListener;
 public class ApiClient {
   public interface Api {}
 
+  //Formatter with fixed length and timezone
+  public static final DateTimeFormatter ISO_FIXED_FORMAT =
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/Java/libraries/feign/ApiClient.mustache]: Issue: this should go in a "java8" block. Fixing it.

**Comment by cbornet** [comment]
I just read from rfc3339
> The offset between local time and UTC is often useful information.
   For example, in electronic mail (RFC2822, [IMAIL-UPDATE]) the local
   offset provides a useful heuristic to determine the probability of a
   prompt response.  Attempts to label local offsets with alphabetic
   strings have resulted in poor interoperability in the past [IMAIL],
   [HOST-REQ].  As a result, RFC2822 [IMAIL-UPDATE] has made numeric
   offsets mandatory.

So I'm not sure about this PR in the end.

**Comment by cbornet** [comment]
On further thought, the best would probably be to expose the date formatter so the user can configure it the way he wants (milli/micro/nano precision, UTC/time-offset, ...). I'll see if that's easily doable.

**Comment by wing328** [comment]
@cbornet totally agree that's a better approach and that's what we did for the C# API client: https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/csharp/SwaggerClient/src/IO.Swagger/Client/Configuration.cs#L288-L290

**Comment by cbornet** [comment]
Closing for now. I'll see how to expose the formatter config.

**Pull request closed without merging**